frontend/src/scenes/dashboard/DashboardHeader.tsx

```diff
@@ -232,9 +234,10 @@ export function DashboardHeader(): JSX.Element | null {
                                             </LemonButton>
                                             {canEditDashboard && (
                                                 <LemonButton
-                                                    onClick={() =>
-                                                        deleteDashboard({ id: dashboard.id, redirect: true })
-                                                    }
+                                                    onClick={() => {
+                                                        setDeleteDashboardValue('dashboardId', dashboard.id)
+                                                        showDeleteDashboardModal()
```

Comment:
Ideally these two actions would be just one `showDeleteDashboardModal(dashboard.id)`, since there's no other _valid_ way to use either. The less invalid ways of using APIs there are, the better for future us and for the product.